src/sentry/static/sentry/app/views/organizationEventsV2/utils.jsx

```diff
@@ -78,3 +79,25 @@ export function eventTagSearchUrl(tag, location) {
     query,
   };
 }
+
+/**
+ * Fetches tag distributions for heatmaps for an array of tag keys
+ *
+ * @param {Object} api
+ * @param {String} orgSlug
+ * @param {Array} tagList
+ * @returns {Promise<Object>}
+ */
+export function fetchTags(api, orgSlug, tagList) {
+  return api
+    .requestPromise(`/organizations/${orgSlug}/events-heatmap/`, {
+      query: {keys: tagList},
+    })
+    .then(resp => {
+      const tags = {};
+      resp.forEach(tag => {
+        tags[tag.key] = tag;
+      });
+      return tags;
```

Comment:
You could use `reduce` here
```javascript
return resp.reduce((result, item) => {
  result[tag.key] = tag;
  return result;
}, {});
```
Might not be worth it though :golf:

Reply:
```
return resp.reduce((result, tag) => ({
  ...result,
  [tag.key]: tag,
}), {});
```
⛳️

Reply:
Lol :). Subjectively, I find forEach to be much more readable and more performant for building an object